### PR TITLE
Add getAngleUpperIndexRaw

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

File: include/geolib/sensors/LaserRangeFinder.h [hunk 1]
```diff
@@ -129,6 +130,21 @@ class LaserRangeFinder {
 
     // Number of beams in a half circle
     uint i_half_circle_;
+    
+    /**
+     * @brief Get the index of the first beam with a higher angle than this beam
```
Comment (PetervDooren) [inline include/geolib/sensors/LaserRangeFinder.h]:
```suggestion
     * @brief Get the index of the first beam with a higher angle than \p angle. Results may lie outside of the sensor range.
```

File: src/sensors/LaserRangeFinder.cpp [hunk 2]
```diff
@@ -64,26 +64,26 @@ void LaserRangeFinder::RenderResult::renderLine(const Vec2& p1, const Vec2& p2)
     // In the latter case, we may need to render two parts
 
     uint i_min1, i_max1, i_min2, i_max2;
```
Comment (PetervDooren) [inline src/sensors/LaserRangeFinder.cpp:66]: would it not be easier to make these regular int rather than do all the casting below?

File: test/test_shape.cpp [hunk 3]
```diff
@@ -27,7 +27,6 @@ TEST_F(ShapeTest, Contains)
     ASSERT_TRUE(shape.contains(max));
 
     ASSERT_TRUE(shape.contains(side_center));
-    std::cout << "Side_center_triangle" << std::endl << std::endl << std::endl << std::endl << std::endl;
```
Comment (PetervDooren) [inline test/test_shape.cpp:30]: ? This seems unrelated to the lrf testcases.

File: include/geolib/sensors/LaserRangeFinder.h [hunk 4]
```diff
@@ -129,6 +130,21 @@ class LaserRangeFinder {
...
+     * @param angle radial angle of the beam
+     * @return Index of the beam
+     */
+    int getAngleUpperIndexRaw(double angle) const;
```
Comment (PetervDooren) [inline include/geolib/sensors/LaserRangeFinder.h:139]: I think these should be public.

**Review by PetervDooren — CHANGES_REQUESTED** [review]

File: src/sensors/LaserRangeFinder.cpp (hunk 2 above)
Comment (MatthijsBurgh) [reply src/sensors/LaserRangeFinder.cpp:66]: These are eventually used to iterate. So you could only reduce the number of casts from 4 to 2. But I think it is better to keep these uint, So it is very clear these shouldn't be negative and therefore not lie outside the sensor range.

File: test/test_shape.cpp (hunk 3 above)
Comment (MatthijsBurgh) [reply test/test_shape.cpp:30]: Yes indeed. But I found it and creating a PR to remove this single line might be too much

File: include/geolib/sensors/LaserRangeFinder.h (hunk 4 above)
Comment (MatthijsBurgh) [reply include/geolib/sensors/LaserRangeFinder.h:139]: Right now, these aren't used outside this class. So I don't see a reason why. We can change it when it is needed.

File: src/sensors/LaserRangeFinder.cpp (hunk 2 above)
Comment (PetervDooren) [reply src/sensors/LaserRangeFinder.cpp:66]: This is a matter of preference. face to face discussion said to leave as is.

**Review by PetervDooren — APPROVED** [review]
